cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go

```diff
@@ -451,27 +454,45 @@ func createTestConfigs(specs ...testSpec) []*testConfig {
 			Kind: config.machineSet.GetKind(),
 			UID:  config.machineSet.GetUID(),
 		}
-
+		config.machineTemplate = &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+				"kind":       machineTemplateKind,
+				"metadata": map[string]interface{}{
+					"name":      "TestMachineTemplate",
+					"namespace": spec.namespace,
+					"uid":       "TestMachineTemplate",
+				},
+			},
+		}
 		if spec.capacity != nil {
 			klog.V(4).Infof("adding capacity to machine template")
-			config.machineTemplate = &unstructured.Unstructured{
-				Object: map[string]interface{}{
-					"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
-					"kind":       machineTemplateKind,
-					"metadata": map[string]interface{}{
-						"name":      "TestMachineTemplate",
-						"namespace": spec.namespace,
-						"uid":       "TestMachineTemplate",
-					},
-				},
-			}
+
 			if err := unstructured.SetNestedStringMap(config.machineTemplate.Object, spec.capacity, "status", "capacity"); err != nil {
 				panic(err)
 			}
 		} else {
 			klog.V(4).Infof("not adding capacity")
 		}
 
+		if spec.nodeInfo != nil {
+			for _, v := range []*unstructured.Unstructured{
+				config.machineSet,
+				config.machineDeployment,
+				config.machineTemplate,
+			} {
+				if v == nil {
+					continue
+				}
+				if err := unstructured.SetNestedStringMap(v.Object, map[string]string{
+					"architecture":    spec.nodeInfo.Architecture,
+					"operatingSystem": spec.nodeInfo.OperatingSystem,
+				}, "status", "nodeInfo"); err != nil {
+					panic(err)
+				}
+			}
+		}
+
```

Comment:
Hi @elmiko, should we add that field only to machineTemplates or also to machineSets and machineDeployments? I don't see status.capacity set for machineSets and machineDeployments, but I also didn't find tests using a non-nil value for capacity.

Reply:
do we have tests in place that use a non-nil capacity and machine templates?